### PR TITLE
Remove Deprecated property ISummaryBaseConfiguration.summarizerClientElection and disable underlying code

### PR DESCRIPTION
**Pull request by NicholasCouri**
## Description

This change removes the deprecated property SummarizerClientElection as it is a scenario that shows several problems like multiple summarizers being run at the same time.  

This change also removes the underlying code as well as removes the override variable from the container runtime arguments. 

[Dev Azure 2561](https://dev.azure.com/fluidframework/internal/_workitems/edit/2561)

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -211,6 +211,7 @@ export interface ISummaryBaseConfiguration {
     initialSummarizerDelayMs: number;
 
     /**
+     * @deprecated
```
Comment (kian-thompson) [inline packages/runtime/container-runtime/src/containerRuntime.ts:214]: Why not deprecate this is `main` and then nuke in this PR?

Comment (NicholasCouri) [reply packages/runtime/container-runtime/src/containerRuntime.ts:214]: I did not know I could  :)  If so, I can, for sure.

Comment (kian-thompson) [reply packages/runtime/container-runtime/src/containerRuntime.ts:214]: See [Communicating breaking changes](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes#breaking-changes-vs-upcoming-changes)
> An important note to remember is that we may announce upcoming changes during a **minor or major** release.

File: BREAKING.md [hunk 2]
```diff
@@ -19,11 +19,11 @@ It's important to communicate breaking changes to our stakeholders. To write a g
 
 ## 2.0.0-internal.3.0.0 Upcoming changes
 - [Deprecated IPendingFlush](#Deprecated-IPendingFlush)
-- [Deprecate ISummaryBaseConfiguration.summarizerClientElection](#Deprecate-ISummaryBaseConfigurationsummarizerClientElection)
+- [Remove ISummaryBaseConfiguration.summarizerClientElection](#Remove-ISummaryBaseConfigurationsummarizerClientElection)
```
Comment (kian-thompson) [inline BREAKING.md:22]: I believe this belongs in the `Breaking changes` section

Comment (NicholasCouri) [reply BREAKING.md:22]: opsss - missed it. Thanks kian!

**Review by kian-thompson — APPROVED** [review]

**Pull request merged**